event-sourcing/event-publisher/event-publisher-process/src/main/java/uk/gov/justice/services/eventsourcing/publishing/EventDeQueuerAndPublisher.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.justice.services.eventsourcing.publishing;
+
+import static java.lang.String.format;
+import static javax.transaction.Transactional.TxType.REQUIRES_NEW;
+
+import uk.gov.justice.services.eventsourcing.publisher.jms.EventPublisher;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.Event;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.EventConverter;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+
+public class EventDeQueuerAndPublisher {
+
+    @Inject
+    EventDeQueuer eventDeQueuer;
+
+    @Inject
+    EventPublisher eventPublisher;
+
+    @Inject
+    EventConverter eventConverter;
+
+    @Inject
+    Logger logger;
+
+    @Transactional(REQUIRES_NEW)
+    public boolean deQueueAndPublish() {
+
+        final Optional<Event> event = eventDeQueuer.popNextEvent();
+        if (event.isPresent()) {
+            logger.debug(format("Publishing event %s", event.get().getName()));
+            final JsonEnvelope jsonEnvelope = eventConverter.envelopeOf(event.get());
+            eventPublisher.publish(jsonEnvelope);
+
+            return true;
+        }
+
+        logger.debug("No event found to publish");
```

Comment:
I suggest dropping this to trace, else the logs will fill up even when no events are being processed